articles/quickstart/spa/vuejs/_includes/_centralized_login.md

```diff
@@ -2,475 +2,441 @@
 
 ```js
-// src/router.js
+import { getInstance } from "./authService";
 
-// .. other imports
-import auth from "./auth/authService";
+export const authGuard = (to, from, next) => {
+  const authService = getInstance();
 
-// .. routes list
+  const fn = () => {
+    // If the user is authenticated, continue with the route
+    if (authService.isAuthenticated) {
+      return next();
+    }
 
-// Existing router declaration
-const router = new Router({
-  mode: 'history',
-  routes
-});
+    // Otherwise, log in
+    authService.loginWithRedirect({ appState: { targetUrl: to.fullPath } });
+  };
 
-// NEW - add a `beforeEach` handler to each route
-router.beforeEach((to, from, next) => {
-  if (to.path === "/" || to.path === "/callback" || auth.isAuthenticated()) {
-    return next();
+  // If loading has already finished, check our auth state using `fn()`
+  if (!authService.loading) {
+    return fn();
   }
 
-  // Specify the current path as the customState parameter, meaning it
-  // will be returned to the application after auth
-  auth.login({ target: to.path });
-});
-
-// Existing export
-export default router;
+  // Watch for the loading property to change before we check isAuthenticated
+  authService.$watch("loading", loading => {
+    if (loading === false) {
+      return fn();
+    }
+  });
+};
 ```
 
-Given this, any page that is not either the home page or the callback URL will cause the application to show the login prompt if the user is not authenticated.
+Notice that in the call to `loginWithRedirect`, the URL that the user was trying to access (`to.fullPath`) is supplied. This URL will be returned to the app after authentication and is used to redirect the user to the place they were trying to reach before they logged in.
+
+:::note
+To see how the `targetUrl` property is used to navigate the user once they have authenticated, revisit the <a href="#create-a-vue-plugin">Create a Vue Plugin</a> section again and inspect how the plugin is installed into your Vue app.
+:::
+
+Finally, open up `src/router.js` and use the guard to protect the `/profile` route, as in the following example:
 
-<%= include('../../_includes/_see_it_in_action.md') %>
+```js
+// .. other imports ..
+
+import { authGuard } from "./auth/authGuard";
+
+export default new Router({
+  mode: "history",
+  base: process.env.BASE_URL,
+  routes: [
+    // .. other routes ..
+    {
+      path: "/profile",
+      name: "profile",
+      component: Profile,
+      beforeEnter: authGuard
+    }
+  ]
+});
+```
+
+:::panel Checkpoint
+Start the application and make sure you are logged out. Now, modify the URL in your browser's navigation bar to include the `/profile` path on the end. This should cause the application to ask you to authentication. Once you have logged in, the profile page should be visible without you having to navigate there manually.
```

Comment:
Could not reach this, see comment above about compile error.

Reply:
Should be ok with the above fixed.